caas/kubernetes/provider/application/application.go

```diff
@@ -1464,7 +1463,13 @@ func (a *app) applicationPodSpec(config caas.ApplicationConfig) (*corev1.PodSpec
 			Image:           config.AgentImagePath,
 			WorkingDir:      jujuDataDir,
 			Command:         []string{"/opt/containeragent"},
-			Args:            []string{"init", "--data-dir", jujuDataDir, "--bin-dir", "/charm/bin"},
+			Args: []string{
+				"init",
+				"--containeragent-pebble-dir", "/containeragent/pebble",
+				"--charm-modified-version", strconv.Itoa(config.CharmModifiedVersion),
+				"--data-dir", jujuDataDir,
+				"--bin-dir", "/charm/bin",
+			},
 			Env: []corev1.EnvVar{
 				{
 					Name:  "JUJU_CONTAINER_NAMES",
```

Comment:
Do we want to add "charm" to the list of container names now that it's running Pebble? AFAIK this list is passed to the uniter / Pebble poller

Reply:
I don't think we need this.